larpandoracontent/LArTwoDReco/LArClusterAssociation/MvaLowEClusterMergingAlgorithm.cc

```diff
@@ -0,0 +1,750 @@
+                            float mag(centroidToHitVec.GetMagnitude());
+                            CartesianVector normCentroidVec((x - centroid.GetX()) / mag, 0.f , (z - centroid.GetZ()) / mag );
+                            float dotProduct(vec.GetDotProduct(normCentroidVec));
+                            //std::cout << " Centroid1: " << centroid1.GetX() << " || " << centroid1.GetZ() <<  " Centroid2: " << centroid2.                              GetX() << " || " << centroid2.GetZ() << pMCParticle << listName << std::endl;
+                            //  std::cout << "CENTROID: " << centroid.GetX() << " || " << centroid.GetZ() << std::endl;
+
+                            if (std::abs(dotProduct) < 0.03 )
+                            {
+                                //std::cout << "Max Mag: " << maxMag << std::endl;
+                                if (maxMag < mag)
+                                {
+                                        maxMag = mag;
+                                        sectorHits.clear();
+                                        hitIsUsed[pCaloHit] = true;
+                                        sectorHits.push_back(pCaloHit);
+                                        //std::cout << "DOT PRODUCT: " << dotProduct << "Magnitude: " << mag << std::endl;
+                                }
+                            }
+                         }
+
+
+
+                }
+                //std::cout << "Last Hit to add: " << sectorHits.size() << std::endl;
+                for (const CaloHit *pCaloHit: sectorHits)
+                {
+                       clusterEdgeHits.push_back(pCaloHit);
+                }
+                //std::cout << "Number of hits found: " << clusterEdgeHits.size() << std::endl;
+                if (clusterCaloHits.size() < clusterEdgeHits.size())
+                {
+                        std::cout << "Error in Edge Hit Algorithm: Incorrect edge hit classification !!!!" << std::endl;
+
+                }
+        }
+
+
+
+
+        return clusterEdgeHits;
+}
+
+//----------------------------------------------------------------------------------------------------
+
+template <typename T>
+float MvaLowEClusterMergingAlgorithm<T>::Distance(const CartesianVector vector1, const CartesianVector vector2) const
+{
+    const float dx{std::abs(vector1.GetX() - vector2.GetX())};
+    const float dz{vector1.GetZ() - vector2.GetZ()};
+    float distance{std::sqrt(dx * dx + dz * dz)};
+
+    return distance;
+}
+
+//-------------------------------------------------------------------------------------------------
+
+template <typename T>
+double MvaLowEClusterMergingAlgorithm<T>::Angle(const CartesianVector vector1, const CartesianVector vector2) const
+{
+    const double dx{vector1.GetX() - vector2.GetX()};
+    const double dz{vector1.GetZ() - vector2.GetZ()};
+    double angle{tan(dx / dz)};
+
+    return angle;
+}
```

Comment:
I think both of these functions can be removed given previous comments on existing CartesianVector functionality